common/djangoapps/third_party_auth/api/tests/test_views.py

```diff
@@ -81,6 +86,57 @@ def setUp(self):
         UserFactory.create(username=CARL_USERNAME, password=PASSWORD)
 
 
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class ProviderViewAPITests(TpaAPITestCase):
+    """
+    Test that the provider endpoint returns the appropriate data
+    """
+    def test_results(self):
```

Comment:
Could you split up this into multiple tests for linked and unlinked users? Also, you can use the ``ddt`` instead of a loop.